Mage/src/main/java/mage/abilities/effects/common/continuous/CastAsThoughItHadFlashAllEffect.java

```diff
@@ -50,25 +49,19 @@ public CastAsThoughItHadFlashAllEffect copy() {
     }
 
     @Override
-    public boolean applies(UUID affectedSpellId, Ability source, UUID affectedControllerId, Game game) {
-        if (anyPlayer || source.isControlledBy(affectedControllerId)) {
-            Card card = game.getCard(affectedSpellId);
+    public boolean applies(UUID objectId, Ability affectedAbility, Ability source, Game game, UUID playerId) {
+        if (affectedAbility instanceof SpellAbility && (anyPlayer||source.isControlledBy(playerId))) {
+            Card card = ((SpellAbility) affectedAbility).getCharacteristics(game);
             if (card != null) {
-                //Allow lands with morph to be played at instant speed
-                if (card.isLand(game)) {
-                    boolean morphAbility = card.getAbilities().stream().anyMatch(MorphAbility.class::isInstance);
-                    if (morphAbility) {
-                        Card cardCopy = card.copy();
-                        cardCopy.removeAllCardTypes(game);
-                        cardCopy.addCardType(game, CardType.CREATURE);
-                        return filter.match(cardCopy, affectedControllerId, source, game);
-                    }
-                }
-                return filter.match(card, affectedControllerId, source, game);
+                return filter.match(card, playerId, source, game);
             }
         }
         return false;
     }
+    @Override
+    public boolean applies(UUID objectId, Ability source, UUID affectedControllerId, Game game) {
```

Comment:
can you explain why this method is being split into two with different signatures?

Reply:
I need access to the affectedAbility argument. The parent class has the full applies method calling the shorter one by default. However, instead of returning false, I should be throwing an IllegalArgumentException.